V2.1 fixes

yeozhishen opened the pull request:
Fixes #140 , #133 , #127, #124, #119

yeozhishen committed:
Modify DG
- Basic syntax changes, change of </br> to ,br.
- remove broken hyperlinks

yeozhishen committed:
Modify Category UML, Expense
Fixed getExpenseFromFile, in the event that a category does not exist,
changes the category field to null. This is done the preserve the data
integrity of the event class when loading

yeozhishen committed:
Modify Main Loop Error message, GeneralInputParser
Modified GeneralInputParser to handle errors in which only spaces were
provided

yeozhishen committed:
Modify UI
Edited UI such that it returns an empty string and recreates the scanner
upon receiving input that causes the scanner class to throw and
exception. This will prevent the main loop exception from being invoked

yeozhishen committed:
Modify expense, AddParser
Limit expenses loaded from file via getExpenseFromFile in Expense and
ParseInput in addparser to be up to 7 digits, exluding the decimal
places. This will make it harder to encounter floating point overflow/truncations errors

STeng618 commented on docs/UserGuide.md:
```diff
@@ -136,6 +161,7 @@ If not provided, will summarise up to end of list <br>
 `/start START_DATE` : Expenses from this `START_DATE` onwards (inclusive) or till the end date (if specified) 
 will be summarised. Must be in the format `dd-MM-yyyy`. <br>
 `/end END_DATE` : Expenses up to this `END_DATE` (inclusive) or from the start date (if specified) will be summarised.
```
I think the repetition is this part

STeng618 reviewed (commented):
lgtm

yeozhishen committed:
Modify Tests
Modify Tests in tandem with the category change, also to make the
category clearer as other tests may use the category FOOD

yeozhishen committed:
Fix 127-incorrect-naming-UG

yeozhishen committed:
Add example outputs for Categories in UG

yeozhishen committed:
Remove duplicate summarize parameters in UG
